lib/src/network_image_mock.dart

```diff
@@ -74,11 +74,28 @@ class MockStreamSubscription<T> extends Mock implements StreamSubscription<T> {}
 MockHttpClient createMockImageHttpClient() {
   final MockHttpClient client = MockHttpClient();
   final MockHttpClientRequest request = MockHttpClientRequest();
+  final MockHttpClientRequest svgRequest = MockHttpClientRequest();
+
+  when(client.getUrl(any)).thenAnswer((Invocation invocation) {
+    return Future<HttpClientRequest>.value(
+      invocation.positionalArguments[0].path.endsWith(".svg")
+          ? svgRequest
+          : request,
+    );
+  });
+
+  mockRequestResponse(request: request, image: image);
+  mockRequestResponse(request: svgRequest, image: svgImage);
+
+  return client;
+}
+
+void mockRequestResponse({
+  required MockHttpClientRequest request,
+  required List<int> image,
+}) {
   final MockHttpClientResponse response = MockHttpClientResponse();
   final MockHttpHeaders headers = MockHttpHeaders();
-
-  when(client.getUrl(any))
-      .thenAnswer((_) => Future<HttpClientRequest>.value(request));
```

Comment:
I assume this can all be simplified into just picking either `image` or `svgImage`. It seems creating two mocked requests and handling them both is redundant, but I might be wrong.

I will definitely test this tomorrow and also the shorter version, but if you do it before me, that would be awesome!

Reply:
Yes, that was the idea I had in mind at first, but I couldn't find a way to detect on the `response.listen()` mock if the request was an svg or png image. Therefor had to create two requests and two different responses.

Reply:
Did you figure out if it's possible to simplify it?

Reply:
I will just merge it.